muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/dao/UserChatSettingsDAO.java

```diff
@@ -0,0 +1,68 @@
+package fi.otavanopisto.muikku.plugins.chat.dao;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import fi.otavanopisto.muikku.plugins.CorePluginsDAO;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatSettings;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatSettings_;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatVisibility;
+import fi.otavanopisto.muikku.plugins.forum.model.ForumMessage;
+import fi.otavanopisto.muikku.schooldata.SchoolDataIdentifier;
+
+
+public class UserChatSettingsDAO extends CorePluginsDAO<UserChatSettings> {
+	
+  private static final long serialVersionUID = -7830619828801454118L;
+
+  public UserChatSettings create(String userIdentifier, UserChatVisibility userChatVisibility) {
+    UserChatSettings settings = new UserChatSettings(userIdentifier, userChatVisibility);
+    getEntityManager().persist(settings);
+    return settings;
+  }
+  
+  public List<UserChatSettings> listAll(String userIdentifier, UserChatVisibility userChatVisibility) {
```

Comment:
Unused method? All DAO classes have a `listAll` method through class inheritance. This one has `userIdentifier` and `userChatVisibility` as arguments but doesn't use them at all and the code essentially does what plain `listAll()` does, anyway.